WordPress/src/main/java/org/wordpress/android/ui/notifications/adapters/NotesAdapter.kt

```diff
@@ -359,29 +375,50 @@ class NotesAdapter(
         @Suppress("ForbiddenComment")
         fun bindInlineActionIconsForNote(note: Note) = Notification.from(note).let { notification ->
             when (notification) {
-                Comment -> {
-                    actionIcon.setImageResource(R.drawable.star_empty)
-                    actionIcon.isVisible = true
-                    actionIcon.setOnClickListener {
-                        // TODO: handle tap on comment's inline action icon (the star)
-                    }
-                }
-                is PostNotification -> {
-                    actionIcon.setImageResource(R.drawable.block_share)
-                    actionIcon.isVisible = true
-                    actionIcon.setOnClickListener {
-                        coroutineScope.launch {
-                            inlineActionEvents.emit(
-                                InlineActionEvent.SharePostButtonTapped(notification)
-                            )
-                        }
-                    }
-                }
+                Comment -> bindLikeCommentAction(note)
+                is PostNotification.NewPost -> bindLikePostAction()
+                is PostNotification -> bindShareAction(notification)
                 is Unknown -> {
                     actionIcon.isVisible = false
                 }
             }
         }
+
+        private fun bindShareAction(notification: PostNotification) {
+            actionIcon.setImageResource(R.drawable.block_share)
+            val color = contentView.context.getColorFromAttribute(R.attr.wpColorOnSurfaceMedium)
+            ImageViewCompat.setImageTintList(actionIcon, ColorStateList.valueOf(color))
+            actionIcon.isVisible = true
+            actionIcon.setOnClickListener {
+                coroutineScope.launch {
+                    inlineActionEvents.emit(
+                        InlineActionEvent.SharePostButtonTapped(notification)
+                    )
+                }
+            }
+        }
+
+        private fun bindLikePostAction() {
+            // implement like post action
+        }
+
+        private fun bindLikeCommentAction(note: Note) {
+            if (note.canLike().not()) return
+
+            actionIcon.isVisible = true
+            actionIcon.setImageResource(if (note.hasLikedComment()) R.drawable.star_filled else R.drawable.star_empty)
+            ImageViewCompat.setImageTintList(actionIcon, null)
```

Comment:
I've noticed a minor UI issue with the visibility of non filled star in _Dark theme_.

![Screenshot_20240215_125559](https://github.com/wordpress-mobile/WordPress-Android/assets/304044/a50e2bb5-9071-4892-a52c-20a58e9cb2bf)

I think removing the `android:fillAlpha="0.6"` from the `star_empty` drawable and adding a tint color would fix this
```suggestion
            if (note.hasLikedComment()) {
                actionIcon.setImageResource(R.drawable.star_filled)
                ImageViewCompat.setImageTintList(actionIcon, null)
            } else {
                actionIcon.setImageResource(R.drawable.star_empty)
                val color = contentView.context.getColorFromAttribute(R.attr.wpColorOnSurfaceMedium)
                ImageViewCompat.setImageTintList(actionIcon, ColorStateList.valueOf(color))
            }
```